### ISSUE TITLE
Feature: IP Logging?

### ISSUE THREAD

**Issue by computernewb** [OpenRCT2/OpenRCT2#4566]
I host two dedicated servers on OpenRCT2, and as one can expect there are frequently people griefing the park and placing waterspouts, raising land, etc everywhere. I have disabled placing scenery and raising land/water but then someone is still spamming paths and demolishing rollercoasters one by one. I think IP logging would be a nice feature, so we can ban anyone who performs these actions.

**Comment by Nubbie** [comment]
Presume the problem for you would be to filter out who did what, my guess is that this should be together with  #4273

**Comment by computernewb** [comment]
The problem isn't that I don't know who it is. I do know who it is. The problem is that the user frequently changes their username and insults/trolls everyone. There is no way to get their IP from what I can tell which means there is no way I can ban them.

**Comment by Nubbie** [comment]
@wolfreak99 Know you had a solution of the problem on your version, maybe something you could like to help out with?

**Comment by computernewb** [comment]
Yeah, if anyone has any solution to the problem, it would be greatly appreciated. Thanks.

**Comment by wolfreak99** [comment]
If you can convince the main devs to have ips in the game, even if it's as much as from a host server logging the ips and usernames in the external console, I'll report my code to it. But good luck on convincing them to have anything ip related. I gave up trying to convince them it was a good idea, and I just see multiplayer as pretty much doomed if not dead anyways.

**Comment by duncanspumpkin** [comment]
IP bans as said multiple times is pointless as it is trivial to beat. Use white lists to prevent spammers from using commands they shouldn't.

**Comment by IntelOrca** [comment]
@duncanspumpkin this is a logging feature, not banning. As @Nubbie correctly mentioned, it could be done as part of #4273.

**Comment by computernewb** [comment]
@duncanspumpkin This is not a request for IP Banning to be added to a feature, but rather IP Logging. Aside from this, I would like my server to be opened to all, as most players aren't gonna wanna go onto a site and register, fill out an application, or what have you. I don't think having whitelists is necessary because it's just one person griefing, not anyone else. I get that multiplayer is not intended for more than 3-4 players (let alone intended for strangers) but I feel like this would be a nice feature to implement. Is IP logging completely not going to be added at all or would it be implemented if a pull request were to be submitted? (Wolf said that the developers were against adding anything ip related). Because if it's allowed then I will take a look at the code and make a contribution for this feature at some point in the future.

**Comment by duncanspumpkin** [comment]
Oh sorry I see ip and multiplayer and just get annoyed.

**Comment by computernewb** [comment]
So this is not a feature that the developers would want added at all then? 

You can use tcpdump although that's not very useful if there is more than one person on the server.

**Comment by duncanspumpkin** [comment]
IP logging is fine we can add that to the logging code when that is done. Right now we are very close to finishing a big chunk of the final drawing code so progress on anything else is slow.

**Comment by computernewb** [comment]
That would be really great. I'm sure me and a lot of other server owners would appreciate it. 

And shoutouts to all OpenRCT2 devs for their hard work and dedication to the project :)

**Comment by PFCKrutonium** [comment]
I'm going to give this a shot, no guarantees though.

**Comment by Margen67** [comment]
#4607 ?

**Comment by janisozaur** [comment]
#4607 has the technical side of it, there's still user-facing reporting that needs to be done.

**Comment by thesemendemon** [comment]
youre gay
